src/Kraken/Config.hs

```diff
@@ -22,13 +23,42 @@ defaultKrakenConfig = KrakenConfig {
   numberOfRetries = 1
  }
 
-loadConfig :: FilePath -> IO (KrakenConfig, Config)
-loadConfig configFile = do
-  config <- load [Required configFile]
-  let custom = subconfig "customConfig" config
-  result <- KrakenConfig <$>
-     lookup config "retryDelay" <*>
-     lookupDefault 1 config "numberOfRetries"
+newtype ConfigPair a = ConfigPair { _unConfigPair :: (KrakenConfig, a) }
+                     deriving (Eq, Show)
+
+instance FromJSON a => FromJSON (ConfigPair a) where
+  parseJSON (Object v) = do
+    delay <- v .: "retryDelay"
+    retries <- v .:? "numberOfRetries" >>= return . fromMaybe 1
+    custom <- v .: "customConfig"
+    return $ ConfigPair (KrakenConfig delay retries, custom)
+
+  parseJSON _ = fail "FromJSON.parseJSON failure for ConfigPair"
```

Comment:
We could also define `ConfigPair` in a way that the `Generic` instance would work. On the other hand, it's nice to have `KrakenConfig` have nice field names (that would clash with `ConfigPair` if we did that). @Fuuzetsu: was that your reasoning?